app-android/src/main/java/com/ustadmobile/port/android/view/CourseDiscussionDetail2Fragment.kt

```diff
@@ -0,0 +1,355 @@
+        set(value) {
+            field = value
+            descriptionRecyclerAdapter?.courseDiscussion = value
+            ustadFragmentTitle = value?.courseDiscussionTitle
+        }
+
+
+
+}
+
+
+
+@Composable
+private fun PostsDetail(
+    posts: List<DiscussionPostWithDetails> = emptyList(),
+    onClickPost: (DiscussionPostWithDetails) -> Unit = {}
+){
+
+    LazyColumn(
+        modifier = Modifier.fillMaxWidth(),
+        state = LazyListState(),
+    ) {
+        itemsIndexed(posts) { _, item ->
+            PostDetailItem(item)
+        }
+    }
+
+}
+
+@Composable
+private fun AddPostButton(onClick: () -> Unit) {
+    Button(
+        shape = RoundedCornerShape(50),
+        onClick = {onClick()},
+        modifier = Modifier
+            .padding(12.dp)
+            .height(45.dp)
+            .width(120.dp),
+        elevation = null,
+        colors = ButtonDefaults.buttonColors(
+            backgroundColor = colorResource(R.color.secondaryColor),
+            contentColor = Color.Transparent,
+            disabledBackgroundColor = Color.Transparent,),
+    ) {
+        Row (
+            horizontalArrangement = Arrangement.End) {
+            Image(
+                painter = painterResource(id = R.drawable.ic_add_black_24dp),
+                contentDescription = null,
+                modifier = Modifier.width(25.dp),
+                colorFilter = ColorFilter.tint(color = Color.Black))
+            Spacer(modifier = Modifier.width(5.dp))
+            Text(
+                text = stringResource(R.string.post),
+                color = Color.Black,
+                textAlign = TextAlign.Center,
+                style = Typography.h5
+            )
+        }
+    }
+}
+
+@Composable
+private fun PostDetailItem(post: DiscussionPostWithDetails){
+    Row(
+        modifier = Modifier
+            .padding(8.dp)
+            .fillMaxWidth(),
+        verticalAlignment = Alignment.CenterVertically
+    ){
+        val context = LocalContext.current
+        val datePosted = remember { DateFormat.getDateFormat(context)
+            .format(Date(post.discussionPostStartDate ?: 0)).toString() }
+
+        val fullName= post.authorPersonFirstNames + " " + post.authorPersonLastName
+        UstadMessageField(
+            imageId = R.drawable.ic_person_black_24dp,
+            mainText = fullName,
+            secondText = post.discussionPostTitle ?: "",
+            thirdText = post.postLatestMessage ?: "",
+            fourthText = datePosted,
+            fifthText = post.postRepliesCount.toString() + " replies",
+            thirdTextIcon = R.drawable.ic_baseline_sms_24
+        )
+    }
+}
+
+
+@Composable
+private fun CourseDiscussionDetailScreen(
+    uiState: CourseDiscussionDetailUiState = CourseDiscussionDetailUiState(),
+    onClickAddPost: () -> Unit = {},
+    onClickPost: (DiscussionPostWithDetails) -> Unit = {},
+    onClickDeletePost: (DiscussionPostWithDetails) -> Unit = {},
+){
+
+    Column(
```

Comment:
This needs to use LazyColumn. The posts themselves will be loaded lazily using PagedList (the replacement for DataSourceFactory). You can use a plain list for now (I still need to add PagedList to door - which will be straightforward).

You can then use something like this:


```
LazyColumn  {
    item {
       //description etc here
    }

    items(
        items = uiState.posts,
        key = { it.postUid }
     ) { post ->
         //item content goes here
     }
}
```